lib/rspec/matchers/built_in/compound.rb

```diff
@@ -36,6 +36,28 @@ def expects_call_stack_jump?
           NestedEvaluator.matcher_expects_call_stack_jump?(matcher_2)
         end
 
+        # @api private
+        # @return [Boolean]
+        def diffable?
+          [matcher_1, matcher_2].any?(&method(:matcher_is_diffable?))
```

Comment:
Honestly, I'd rather it be this:

``` ruby
def diffable?
  matcher_is_diffable?(matcher_1) || matcher_is_diffable?(matcher_2)
end
```

I don't think that's enough duplication to put it in an array and use `any?`, particularly when the `Array#any?` way has an extra object allocation for the array.